.github/workflows/step_tests-pip.yml

```diff
@@ -1,54 +1,69 @@
 name: test-pip
-run-name: Run tests using Pip
+run-name: Run main tests using Pip
 
 on:
   workflow_call:
+    inputs:
+      coverage:
+        type: boolean
+        description: Upload coverage to CodeCov
+        default: true
 
-concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}-pip-test
-  cancel-in-progress: true
+permissions:
+  contents: read
 
 jobs:
   test-pip:
+    name: >
+      🐍 ${{ matrix.python-version }}
+      ${{ matrix.dependency_type && format('({0})', matrix.dependency_type) }}
+      ${{ matrix.markdown-it-py && format('(markdown-it-py {0})', matrix.markdown-it-py) }}
+      ${{ matrix.quarto && '(Quarto)' }}
+      ${{ matrix.no_kernel && '(No kernel)' }}
     runs-on: ubuntu-latest
 
     strategy:
+      fail-fast: false
       matrix:
         python-version: [ "3.8", "3.9", "3.10", "3.11", "3.12"]
-        markdown-it-py-version: ["~=3.0"]
-        pip-flags: [""]
         include:
+          # Test pre-release versions
           - python-version: "3.x"
-            pip-flags: "--pre --upgrade --upgrade-strategy eager"
+            dependency_type: "pre"
+          # Test minimum markdown-it-py supported (otherwise the most recent version is used)
           - python-version: "3.x"
-            markdown-it-py-version: "~=2.0"
+            markdown-it-py: "~=2.0"
           - python-version: "3.x"
-            markdown-it-py-version: ""
+            markdown-it-py: false
           - python-version: "3.x"
             no_kernel: true
           - python-version: "3.x"
             quarto: true
 
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
       - name: Base Setup
         uses: jupyterlab/maintainer-tools/.github/actions/base-setup@v1
         with:
           python_version: ${{ matrix.python-version }}
+          dependency_type: ${{ matrix.dependency_type }}
 
       - name: Install from source
-        run: python -m pip install -e '.[test-cov,test-external]' jupyterlab ${{ matrix.markdown-it-py-version && format('markdown-it-py{0}', matrix.markdown-it-py-version) }} ${{ matrix.pip-flags }}
+        run: >
+          python -m pip install
+          -e '.[test-cov,test-external]'
+          jupyterlab
+          ${{ matrix.markdown-it-py && format('markdown-it-py{0}', matrix.markdown-it-py) || '' }}
 
       - name: Install a Jupyter Kernel
         if: ${{ !matrix.no_kernel }}
         run: python -m ipykernel install --name python_kernel --user
-
       - name: Uninstall markdown-it-py
         # Markdown-it-py is a dependency of Jupytext,
         # but Jupytext should still work if it is not installed
-        if: ${{ matrix.markdown-it-py-version == '' }}
+        if: ${{ matrix.markdown-it-py == 'false' }}
```

Comment:
Maybe this should be `== false`? Maybe I'll switch to using a flag like `no_kernel` or `quarto` which works and also has the advantage of being more explicit.

Reply:
Yes you're right. Github workflow is annoying like that where sometimes the variable is converted to a string, sometimes it's the original json type. In the case of matrix it's the latter.

Reply:
FYI I have added an additional `no_markdown-it-py` (false/true) at https://github.com/mwouts/jupytext/pull/1204